lib/ecto/query/builder.ex

```diff
@@ -544,6 +555,21 @@ defmodule Ecto.Query.Builder do
   end
 
   @doc """
+  Called by escaper at runtime to verify binaries.
+  """
+  def binary!(v) do
+    unless is_binary(v) do
+      raise ArgumentError, "`unsafe_fragment/1` expects only an interpolated string. " <>
```

Comment:
This error message is quote good but I think we can probably just say `unsafe_fragment/1 expects a string at compile time` and move the bulk of it to the error message I mentioned above!